MCprep_addon/materials/default_materials.py

```diff
@@ -0,0 +1,216 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+
+import os
+
+import bpy
+
+from .. import conf
+from .. import tracking
+from .. import util
+from . import sync
+
+def default_material_in_sync_library(default_material, context):
+	"""Returns true if the material is in the sync mat library blend file"""
+	if conf.material_sync_cache is None:
+		sync.reload_material_sync_library(context)
+	if util.nameGeneralize(default_material) in conf.material_sync_cache:
+		return True
+	elif default_material in conf.material_sync_cache:
+		return True
+	return False
+
+def sync_default_material(context, material, default_material, engine):
+    """
+    This is ideitical to the normal sync materials function but it also copies
+    the material and changes the name
+    """
+    if default_material in conf.material_sync_cache:
+        import_name = default_material
+    elif util.nameGeneralize(default_material) in conf.material_sync_cache:
+        import_name = util.nameGeneralize(default_material)
+
+    # if link is true, check library material not already linked
+    sync_file = sync.get_sync_blend(context)
+    
+    # I'm no Python expert (I prefer C++) but Google tells me that this means to copy all the elements 
+    # -StandingPad
+    init_mats = bpy.data.materials[:]
+    util.bAppendLink(os.path.join(sync_file, "Material"), import_name, False) # no linking
+
+    imported = set(bpy.data.materials[:]) - set(init_mats)
+    if not imported:
+        return "Could not import " + str(material.name)
+    new_material = list(imported)[0]
+    
+    # checking if there's a node with the label Texture
+    new_material_nodes = new_material.node_tree.nodes
+    if not new_material_nodes.get("Default Texture"):
```

Comment:
See if we can follow the MCprep convention that already exists for "tagging" the nodes within the material to get the diffuse image. Agreed that this first pass does not need to support PBR, but good to try and follow the convention so it's a more natural upgrade later: https://github.com/TheDuckCow/MCprep/blob/master/MCprep_addon/materials/generate.py#L474